tag-historian/nitaghistorian.yml

```diff
@@ -316,12 +256,6 @@ definitions:
               properties:
                 note: This value is an outlier and needs investigation.
               expirationDate: '2020-03-13T09:51:38Z'
```

Comment:
We should remove this one

Reply:
I removed the expirationDate references.

I think "type" could still be useful for clients. Tag viewer shows historical tag values for "string" tags in a table instead of a graph, for example, and it seems less than ideal for clients like that to have to query the tag service for the type info, or detect the type of a value. It isn't possible to change the type of a tag, so it seems OK to attach type info to the values. Any thoughts?

Reply:
It is just a weird mix between historical values and metadata. It would be better to keep responsibilities between tag service and tag historian clear. It also leads to lot of redundant type information for every single value. I would expect clients to know what data a certain tag produces. For example, the tag viewer already retrieves the tag metadata first

Reply:
Paul and I's thinking is that the query-history response isn't useful without the type information. The values will look like strings, and might actually just be strings. Requiring clients to be stateful and keep a map of tag paths to types to be able to interpret the values seems inconvenient. Carson tells me that dashboard builder would eventually like to use the type information in the response as well.

Reply:
Maybe we should provide the type information outside of 'values' so that we just provide the info once and don't repeat it 1000 times.

Reply:
I agree that would save some space. It seems like we could provide a dictionary where the keys are tag paths and the values are types. That would be another breaking change, so I'll take a look at the impact that would have on tag viewer and push proposed changes for what that would look like.